src/Verify.MSTest/Verifier.cs

```diff
@@ -0,0 +1,138 @@
+namespace VerifyMSTest;
+
+public static partial class Verifier
+{
+    static Task AddFile(FilePair path, bool autoVerify)
+    {
+        var context = CurrentTestContext.Value;
+        if (context != null)
+        {
+            var fileName = autoVerify ? path.VerifiedPath : path.ReceivedPath;
+            context.AddResultFile(fileName);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    static Verifier()
+    {
+        VerifierSettings.OnFirstVerify((pair, _, autoVerify) => AddFile(pair, autoVerify));
+        VerifierSettings.OnVerifyMismatch((pair, _, autoVerify) => AddFile(pair, autoVerify));
+    }
+
+    public static readonly AsyncLocal<TestContext?> CurrentTestContext = new();
+
+    static InnerVerifier BuildVerifier(VerifySettings settings, string sourceFile, bool useUniqueDirectory)
+    {
+        var typeName = CurrentTestContext.Value?.FullyQualifiedTestClassName;
+        var type = FindType(typeName.AsSpan());
+
+        if (useUniqueDirectory)
+        {
+            settings.UseUniqueDirectory();
+        }
+
+        var testName = CurrentTestContext.Value?.TestName;
+        if (testName == null)
+        {
+            throw new("TestContext.TestName is null. Ensure being used inside a test");
+        }
+
+        var testNameSpan = testName.AsSpan();
+        var indexOf = testNameSpan.IndexOf('(');
+        if (indexOf > 0)
+        {
+            testNameSpan = testNameSpan[..indexOf];
+        }
+
+        indexOf = testNameSpan.IndexOf('.');
+        if (indexOf > 0)
+        {
+            testNameSpan = testNameSpan[(indexOf + 1)..];
+        }
+
+        VerifierSettings.AssignTargetAssembly(type.Assembly);
+        var method = FindMethod(type, testNameSpan);
+
+        var pathInfo = GetPathInfo(sourceFile, type, method);
+        return new(
+            sourceFile,
+            settings,
+            type.NameWithParent(),
+            method.Name,
+            method.ParameterNames(),
+            pathInfo);
+    }
+
+    static Type FindType(ReadOnlySpan<char> typeName)
```

Comment:
@SimonCropp , if we switch MSTest to the other DerivePathInfo form that takes strings for the type and method name instead of `Type` and `MethodInfo`, we can avoid all this assembly scanning. However, that's presumably a breaking change for consumers.

Should we keep the signature the same for compatibility, or change for performance? If we need to keep it the same, I'll add some caching here to prevent the repeated reflection lookups.

Reply:
happy to make this a breaking change

lets go with strings. if someone has legit requirements for type and method, we can revisit at that point

Reply:
After several hours I wasn't able to find a combination of information directly available on the `TestContext` that matched the behavior before and passed the existing tests.

As a result, I left in the assembly scanning as a fallback, but first try to grab the type from the `TestContext` internals using reflection. I dislike that solution but wasn't able to find anything I liked better. Open to suggestions here.